Marlin/Marlin_main.cpp

```diff
@@ -5428,55 +5428,73 @@ inline void gcode_M303() {
 
 #endif // SCARA
 
-#if ENABLED(EXT_SOLENOID)
-
-  void enable_solenoid(uint8_t num) {
-    switch(num) {
-      case 0:
-        OUT_WRITE(SOL0_PIN, HIGH);
-        break;
-        #if HAS_SOLENOID_1
-          case 1:
-            OUT_WRITE(SOL1_PIN, HIGH);
-            break;
-        #endif
-        #if HAS_SOLENOID_2
-          case 2:
-            OUT_WRITE(SOL2_PIN, HIGH);
-            break;
-        #endif
-        #if HAS_SOLENOID_3
-          case 3:
-            OUT_WRITE(SOL3_PIN, HIGH);
-            break;
-        #endif
-      default:
-        SERIAL_ECHO_START;
-        SERIAL_ECHOLNPGM(MSG_INVALID_SOLENOID);
-        break;
-    }
-  }
-
-  void enable_solenoid_on_active_extruder() { enable_solenoid(active_extruder); }
+#if ENABLED(PNEUMATICS)
 
   void disable_all_solenoids() {
-    OUT_WRITE(SOL0_PIN, LOW);
-    OUT_WRITE(SOL1_PIN, LOW);
-    OUT_WRITE(SOL2_PIN, LOW);
-    OUT_WRITE(SOL3_PIN, LOW);
+    #if HAS_SOLENOID_0
+      OUT_WRITE(SOL0_PIN, LOW);
+    #endif
+    #if HAS_SOLENOID_1
+      OUT_WRITE(SOL1_PIN, LOW);
+    #endif
```

Comment:
I don't see a reason to handle them, unless you think we will need control of four solenoids in the future.

I meant to remove any mention of `SOL2` and `SOL3`.